spring-data-jpa/src/test/java/org/springframework/data/jpa/repository/query/EclipseLinkMetaAnnotatedQueryMethodIntegrationTests.java

```diff
@@ -74,16 +75,12 @@ class EclipseLinkMetaAnnotatedQueryMethodIntegrationTests {
 	@Autowired RoleRepositoryWithMeta repository;
 
 	private static final ResourceLoader RESOURCE_LOADER = new DefaultResourceLoader();
-	private static final String LOG_FILE = "test-eclipselink-meta.log";
+	private static String LOG_FILE;
 
-	@BeforeEach
-	void cleanoutLogfile() throws IOException {
-		new FileOutputStream(LOG_FILE).close();
-	}
-
-	@AfterAll
-	static void deleteLogfile() throws IOException {
-		FileSystemUtils.deleteRecursively(Path.of(LOG_FILE));
+	@BeforeAll
+	static void createLogfile() throws IOException {
+		Path logFile = Files.createTempFile("test-eclipselink-meta", ".log");
+		LOG_FILE = logFile.toAbsolutePath().toString();
```

Comment:
how about making use of junits `@TempDir` leveraging automatic cleanup.

Reply:
Hi, thank you for the reply. Unfortunately with `@TempDir` there is the same problem...  

```java
@TempDir
static Path TMP;

    @BeforeAll
    static void createLogfile() throws IOException {
        Path logFile = TMP.resolve("test-eclipselink-meta.log");
        LOG_FILE = logFile.toAbsolutePath().toString();
    }
```

![image](https://github.com/spring-projects/spring-data-jpa/assets/57438193/76ccca90-5d1a-4a28-b0a4-b0341591a49d)

Reply:
Generally switching to `@TempDir` is generally a good thing but we cannot fix the underlying problem that Eclipselink doesn't close the underlying logger. The offending code is in `EntityManagerSetupImpl`.

Reply:
@mp911de Let me know how I can resolve this issue, if my proposal to use temporary files is ok or if you prefer some other changes. I want to contribute, but I can't check my contribution without run the entire test suite. In the meantime, after further analysis, I can for sure open an issue, or a PR to the EclipseLink project. Thank you very much.